src/libraries/Microsoft.Extensions.DependencyInjection/src/ServiceLookup/ILEmit/ILEmitResolverBuilder.cs

```diff
@@ -96,12 +96,12 @@ private GeneratedMethod BuildTypeNoCache(ServiceCallSite callSite)
             ILGenerator ilGenerator = dynamicMethod.GetILGenerator(512);
             ILEmitResolverBuilderRuntimeContext runtimeContext = GenerateMethodBody(callSite, ilGenerator);
 
-#if SAVE_ASSEMBLIES
+#if SAVE_ASSEMBLIES && NETFRAMEWORK
             var assemblyName = "Test" + DateTime.Now.Ticks;
+            var fileName = assemblyName + ".dll";
```

Comment:
Can you revert these unrelated changes? In general it is best to keep your proposed changes to a single concern. These kinds of changes are distracting from the main intent of the PR.

Reply:
Sure thing 👍 
Should I also split this PR into two? One for each issue fix?

Reply:
Honestly, yes, it might be better to split them into 2 PRs. (3 if you count the SAVE_ASSEMBLIES clean up you were doing here.)

We've had a few regressions in this kinds of changes where we've had to investigate issues. It is easier to narrow down which change caused the problem when unrelated changes are merged separately.

Reply:
No problem, I definitely understand the problems it can cause. Sorry for it, I will separate the changes them. This PR will contain only fixes to #42160. I will try to keep next ones scoped to single a purpose :)